### ISSUE TITLE
Style/CombinableLoops triggered incorrectly

### ISSUE THREAD

**Issue by futuretap** [rubocop/rubocop#13144]
When the first loop calculates an intermediate result that the second loop depends on, the cop should not trigger:

```
strings = %w[hamburg frankfurt munich]
longest = nil
minimum = 0
strings.each do |string|
  if string.size > minimum
    minimum = string.size
    longest = string
  end
end

strings.each do |string|
  puts "#{string} - #{longest}" if longest != string
end
```

--------

## Expected behavior

rubocop should not trigger.

## Actual behavior

```
 C: [Correctable] Style/CombinableLoops: Combine this loop with the previous loop.
    strings.each do |string| ...
```
## RuboCop version

```
1.65.1 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-performance 1.21.1
```

**Comment by Earlopain** [comment]
This is what the safety section is about: https://docs.rubocop.org/rubocop/cops_style.html#safety-11

I do think rubocop can do better here by checking if the second loop accesses a variable that the first one writes to. It will never be fully safe since methods can have side-effects but that doesn't seem very common for such a construct.

**Comment by futuretap** [comment]
keep open
